scijava/scijava-ops-engine/src/test/java/org/scijava/ops/engine/OpMethodDependencyPositionTest.java

```diff
@@ -17,82 +21,93 @@
+			List<Long> out //
+	) {
 		out.clear();
 		for (String s : in)
 			out.add(op.apply(s));
 	}
 
 	@Test
-	public void testOpDependencyAfterOutput() {
-		List<String> in = new ArrayList<>();
-		in.add("1");
-		List<Long> out = new ArrayList<>();
-		ops.op("test.dependencyAfterOutput").arity1().input(in).output(out).compute();
-		List<Long> expected = Arrays.asList(1l);
-		assertIterationsEqual(expected, out);
+	public void testOpDependencyAfter() throws NoSuchMethodException {
+		var m = this.getClass().getDeclaredMethod(//
+				"badDep", //
+				List.class, //
+				Function.class, //
+				List.class //
+		);
+		var info = new OpMethodInfo( //
+					m, //
+					Computers.Arity1.class, //
+					new DefaultHints(), //
+					"test.dependencyAfterInput" //
+		);
+		ValidityException exc = info.getValidityException();
+		String expMsg = "java.lang.IllegalArgumentException: Op Dependencies in " +
+			"static methods must come before any other parameters!";
+		Assertions.assertEquals(expMsg, exc.problems().get(0).getMessage());
 	}
 
-	@OpField(names = "test.squareList")
-	public final Computers.Arity1<List<Long>, List<Long>> squareOp = (in,
-		out) -> {
+	public static void goodThenBadDep( //
+			@OpDependency(name = "someDep") Function<String, Long> op, //
+			List<String> in, //
+			@OpDependency(name = "someOtherDep") Function<String, Long> op2, //
+			List<Long> out //
+	) {
 		out.clear();
-		for (Long l : in)
-			out.add(l * l);
-	};
-
-	@OpMethod(names = "test.dependencyBeforeAndAfterOutput",
-		type = Computers.Arity1.class)
-	public static void OpMethodBar(List<String> in, @OpDependency(
-		name = "test.stringToLong") Function<String, Long> op1, List<Long> out,
-		@OpDependency(
-			name = "test.squareList") Computers.Arity1<List<Long>, List<Long>> op2)
-	{
-		List<Long> temp = new ArrayList<>();
 		for (String s : in)
-			temp.add(op1.apply(s));
-		out.clear();
-		op2.compute(temp, out);
+			out.add(op.apply(s));
 	}
 
 	@Test
-	public void testOpDependencyBeforeAndAfterOutput() {
-		List<String> in = new ArrayList<>();
-		in.add("2");
-		List<Long> out = new ArrayList<>();
-		ops.op("test.dependencyBeforeAndAfterOutput").arity1().input(in).output(out).compute();
-		List<Long> expected = Arrays.asList(4l);
-		assertIterationsEqual(expected, out);
+	public void testOpDependencyBeforeAndAfter() throws NoSuchMethodException {
+		var m = this.getClass().getDeclaredMethod(//
+				"goodThenBadDep", //
+				Function.class, //
+				List.class, //
+				Function.class, //
+				List.class //
+		);
+		var info = new OpMethodInfo( //
+				m, //
+				Computers.Arity1.class, //
+				new DefaultHints(), //
+				"test.dependencyBeforeAndAfterInput" //
+		);
+		ValidityException exc = info.getValidityException();
+		String expMsg = "java.lang.IllegalArgumentException: Op Dependencies in " +
+				"static methods must come before any other parameters!";
+		Assertions.assertEquals(expMsg, exc.problems().get(0).getMessage());
```

Comment:
@gselzer String comparison is a fragile mechanism to tie a test to. It is an opening for this test to fail even though behavior is still correct.

To me it is a sign that something is fundamentally missing that needs to be addressed in https://github.com/scijava/scijava/issues/25 or https://github.com/scijava/scijava/issues/105 (and maybe those issues could be merged?)